src/plugins/controls/public/control_group/editor/create_control.tsx

```diff
@@ -51,12 +51,13 @@ export const CreateControlButton = ({
   updateDefaultGrow,
 }: CreateControlButtonProps) => {
   // Controls Services Context
-  const { overlays, controls } = pluginServices.getServices();
-  const { getControlTypes, getControlFactory } = controls;
-  const { openFlyout, openConfirm } = overlays;
+  const { overlays, controls, theme } = pluginServices.getHooks();
+  const { getControlTypes, getControlFactory } = controls.useService();
+  const { openFlyout, openConfirm } = overlays.useService();
+  const themeService = theme.useService();
 
   const createNewControl = async () => {
-    const PresentationUtilProvider = pluginServices.getContextProvider();
+    const ControlsServicesProvider = pluginServices.getContextProvider();
```

Comment:
I assume there **should** be a way to somehow pass this on to the `ControlEditor`, since the `CreateControlButton` is wrapped in the context provider as part of the `getCreateControlButton` method in `control_group_container.tsx`... But I'm not sure how? If I get rid of this wrapper then, because the `ControlEditor` component relies on the service provider, it crashes once the flyout opens.

Reply:
I think this might be an issue with the flyout instance. As soon as you open a flyout with the core overlays service, it's a totally different react tree, so any context in the react tree above gets lost.

Reply:
Ahhhhh gotchya gotchya, that makes sense 👍  So I guess this is the best way to do it, then - boooo. I was hoping I could make this a little less ugly, heh :)